test/react/navbar.test.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from "react";
+import MyNavbar from "../../app/javascript/components/my_navbar";
+import { expect, test } from "@jest/globals";
+import { getByTestId, queryByText, getByText, render, RenderResult } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import "@testing-library/jest-dom";
+
+const renderNavbar = (
+  loggedIn = true,
+  regPath = "/signup",
+  homePath = "/"
+) : RenderResult => {
+  return render(
+    <MyNavbar loggedIn= {loggedIn} regPath={regPath} homePath={homePath} />
+  );
+};
+
+test("Component renders as expected", () => {
+  const { container } = renderNavbar();
+  expect(container).toMatchSnapshot();
+});
+
+test("Title button has the right link", () => {
+  const { container } = renderNavbar();
+  expect(getByTestId(container, "home-button")).toHaveProperty("href", window.location.href);
```

Comment:
Hmmmm sneaky! This does come with it's own set of issues (what if you were clicking the navbar from the /signups page for instance!) but it definitely works for the time being. Nice work around.

Reply:
I dont understand why the example you gave wouldn't work? From what I've learned is that `window.location.href` gives the url before the ' / ' 
So clicking from " localhost:3000/signup " would still return just "localhost:3000 " 
Am I wrong in thinking this?